### PR TITLE
Ranger Discovery Bundle and Lib Version upgrades

### PR DESCRIPTION
**Pull request by koushikr**
- Moved discovery bundle from [here](https://github.com/appform-io/dropwizard-service-discovery.)
- Updated to dropwizard version 2.1.10 : BOM update. 
- Upgraded to java source, target version 17 with release in 11
- Removed com.fasterxml and introduced dropwizard-jackson for the jackson bindings.
- Upgraded to junit 5 and fixed the necessary tests including wiremock versions and tests. (100+ files changed because of this)
- Upgraded the curator framework version
- Done sonar fixes after the java version upgrade
- Updated the release version
- Updated CHANGELOG.md
- Updated the build.yml to use java17
- Removed logback dependency since dropwizard already has it. 
- Fixed the maven javadoc version to 3.6.3. For [this](https://github.com/apache/maven-javadoc-plugin/commit/d770460b765d2a3ccb9594a1a016c6700501861f)

Callouts

- Can't do much about the netty-handler vulnerability, even the latest netty version is burdened by it. 
- Can't also do much about the snakeyaml vulnerability. Seems like dw 2.1.10 is not compatible with snakeyaml 2.x.

### CODE REVIEW

File: .github/workflows/build-check.yml [hunk 1]
```diff
@@ -15,11 +15,11 @@ jobs:
...
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up JDK 8
+    - uses: actions/checkout@v3
+    - name: Set up JDK 11
       uses: actions/setup-java@v1
       with:
-        java-version: '1.8'
+        java-version: 11
```
Comment (santanusinha) [inline .github/workflows/build-check.yml:22]: Let us move this to 17 directly.

File: ranger-zookeeper/pom.xml [hunk 2]
```diff
@@ -5,7 +5,7 @@
     <parent>
         <artifactId>ranger</artifactId>
         <groupId>io.appform.ranger</groupId>
-        <version>1.0-RC14</version>
+        <version>1.0-RC15</version>
```
Comment (santanusinha) [inline ranger-zookeeper/pom.xml:8]: Introduce dependencyManagement section and add junit-bom dependency

File: ranger-core/src/test/java/io/appform/ranger/core/serviceprovider/ServiceProviderTest.java [hunk 3]
```diff
@@ -98,29 +98,37 @@ protected NodeDataSink<T, TestSerializer<T>> dataSink(Service service) {
...
+    @Test
+    void testInvalidServiceProvider() {
+        try {
+            new TestServiceProviderBuilder<>()
+                    .withServiceName("test-service")
+                    .withNamespace("test")
+                    .withHostname("localhost-1")
+                    .withPort(9000)
+                    .build();
+        } catch (NullPointerException exception) {
```
Comment (santanusinha) [inline ranger-core/src/test/java/io/appform/ranger/core/serviceprovider/ServiceProviderTest.java:110]: You can use `Assertions.assertThrowsExactly(NullPointerException.class, () -> {});`

Comment (santanusinha) [reply ranger-core/src/test/java/io/appform/ranger/core/serviceprovider/ServiceProviderTest.java:110]: Fix in all places

File: ranger-core/src/test/java/io/appform/ranger/core/serviceprovider/ServiceProviderTest.java [hunk 4]
```diff
@@ -98,29 +98,37 @@ protected NodeDataSink<T, TestSerializer<T>> dataSink(Service service) {
...
+    void testInvalidServiceProviderNoHealthCheck() {
+        try {
+            new TestServiceProviderBuilder<>()
+                    .withServiceName("test-service")
+                    .withNamespace("test")
+                    .withHostname("localhost-1")
+                    .withPort(9000)
+                    .withSerializer(new TestSerializerImpl())
+                    .build();
+        } catch (Exception exception) {
```
Comment (santanusinha) [inline ranger-core/src/test/java/io/appform/ranger/core/serviceprovider/ServiceProviderTest.java:125]: Fix

File: ranger-discovery-bundle/src/main/java/io/appform/ranger/discovery/bundle/id/IdGenerator.java [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.appform.ranger.discovery.bundle.id;
+
+import com.google.common.base.Preconditions;
```
Comment (santanusinha) [inline ranger-discovery-bundle/src/main/java/io/appform/ranger/discovery/bundle/id/IdGenerator.java:20]: We should move the whole id generation and other logic out to another module.. this probably has no dependency on dw, only on zk.

Comment (santanusinha) [reply ranger-discovery-bundle/src/main/java/io/appform/ranger/discovery/bundle/id/IdGenerator.java:20]: This means the constrains etc everything

File: ranger-discovery-bundle/src/test/java/io/appform/ranger/discovery/bundle/ServiceDiscoveryBundleHierarchicalSelectorTest.java [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.appform.ranger.discovery.bundle;
```
Comment (santanusinha) [inline ranger-discovery-bundle/src/test/java/io/appform/ranger/discovery/bundle/ServiceDiscoveryBundleHierarchicalSelectorTest.java:18]: Tests seems to be placed in a flat structure. Please move them into proper places

File: ranger-http-client/src/test/java/io/appform/ranger/client/http/BaseRangerHttpClientTest.java [hunk 7]
```diff
@@ -70,21 +73,21 @@ public void startTestCluster() throws Exception {
...
+                .port(wireMockExtension.getPort())
                 .connectionTimeoutMs(30_000)
                 .operationTimeoutMs(30_000)
                 .build();
         log.debug("Started http subsystem");
     }
 
-    @After
+    @AfterEach
     public void stopTestCluster() {
```
Comment (santanusinha) [inline ranger-http-client/src/test/java/io/appform/ranger/client/http/BaseRangerHttpClientTest.java:91]: lol .. rename

File: ranger-http-server/pom.xml [hunk 8]
```diff
@@ -24,23 +20,12 @@
         <dependency>
             <groupId>io.dropwizard</groupId>
             <artifactId>dropwizard-core</artifactId>
-            <version>${dropwizard.version}</version>
         </dependency>
         <dependency>
```
Comment (santanusinha) [inline ranger-http-server/pom.xml:24]: We can remove this, dw comes with logback

File: ranger-server-bundle/src/test/java/io/appform/ranger/server/bundle/RangerServerBundleTest.java [hunk 9]
```diff
@@ -49,15 +45,15 @@
 
 public class RangerServerBundleTest {
 
-    private final JerseyEnvironment jerseyEnvironment = mock(JerseyEnvironment.class);
-    private final MetricRegistry metricRegistry = mock(MetricRegistry.class);
-    private final LifecycleEnvironment lifecycleEnvironment = new LifecycleEnvironment(metricRegistry);
-    private final Environment environment = mock(Environment.class);
-    private final Bootstrap<?> bootstrap = mock(Bootstrap.class);
-    private final Configuration configuration = mock(Configuration.class);
+    private static final JerseyEnvironment jerseyEnvironment = mock(JerseyEnvironment.class);
```
Comment (santanusinha) [inline ranger-server-bundle/src/test/java/io/appform/ranger/server/bundle/RangerServerBundleTest.java:48]: rename to UPPER_SNAKE format for static members

File: ranger-zk-server/config/local.yml [hunk 10]
```diff
@@ -1,14 +1,13 @@
 name: ranger-server
-initialRotationStatus: false
 
 rangerConfiguration:
-  namespace: test
-  zookeeper: localhost:2181
+  namespace: phonepe
```
Comment (santanusinha) [inline ranger-zk-server/config/local.yml:4]: revert

File: ranger-zk-server/config/local.yml [hunk 11]
```diff
@@ -1,14 +1,13 @@
...
+  zookeeper: stg-appzk001.phonepe.nb6:2181,stg-appzk002.phonepe.nb6:2181,stg-appzk003.phonepe.nb6:2181
```
Comment (santanusinha) [inline ranger-zk-server/config/local.yml:5]: revert

File: ranger-zk-server/config/local.yml [hunk 12]
```diff
@@ -1,14 +1,13 @@
...
   disablePushUpdaters: true
 
 server:
-  maxThreads: 128
-  minThreads: 128
+  maxThreads: 1024
```
Comment (santanusinha) [inline ranger-zk-server/config/local.yml:9]: revert

**Review by santanusinha — CHANGES_REQUESTED** [review]

File: ranger-discovery-bundle/src/main/java/io/appform/ranger/discovery/bundle/id/IdGenerator.java (hunk 5 above)
Comment (koushikr) [reply ranger-discovery-bundle/src/main/java/io/appform/ranger/discovery/bundle/id/IdGenerator.java:20]: Decision is to move this to a separate project, and use it here instead. Not changing it now. Resolving the thread

File: ranger-http-client/src/test/java/io/appform/ranger/client/http/BaseRangerHttpClientTest.java (hunk 7 above)
Comment (koushikr) [reply ranger-http-client/src/test/java/io/appform/ranger/client/http/BaseRangerHttpClientTest.java]: This is actually not needed. There is nothing happening afterEach, renamed the beforeEach to prepareHttpMocks.

File: ranger-http-server/pom.xml (hunk 8 above)
Comment (koushikr) [reply ranger-http-server/pom.xml]: Done.

File: ranger-zk-server/config/local.yml (hunk 10 above)
Comment (koushikr) [reply ranger-zk-server/config/local.yml]: Done

File: ranger-zk-server/config/local.yml (hunk 12 above)
Comment (koushikr) [reply ranger-zk-server/config/local.yml]: Done.

File: ranger-zk-server/config/local.yml (hunk 11 above)
Comment (koushikr) [reply ranger-zk-server/config/local.yml]: Done.

File: ranger-server-bundle/src/test/java/io/appform/ranger/server/bundle/RangerServerBundleTest.java (hunk 9 above)
Comment (koushikr) [reply ranger-server-bundle/src/test/java/io/appform/ranger/server/bundle/RangerServerBundleTest.java]: Done.

File: ranger-discovery-bundle/src/test/java/io/appform/ranger/discovery/bundle/ServiceDiscoveryBundleHierarchicalSelectorTest.java (hunk 6 above)
Comment (koushikr) [reply ranger-discovery-bundle/src/test/java/io/appform/ranger/discovery/bundle/ServiceDiscoveryBundleHierarchicalSelectorTest.java:18]: The test is not for HST, it is for SDBundle with HST, similar to SDBundleRotationTest, SDBundleDwMonitorTest and so on. The HST test is properly located in its own package.

File: ranger-core/src/test/java/io/appform/ranger/core/serviceprovider/ServiceProviderTest.java (hunk 4 above)
Comment (koushikr) [reply ranger-core/src/test/java/io/appform/ranger/core/serviceprovider/ServiceProviderTest.java]: Done.

File: .github/workflows/build-check.yml (hunk 1 above)
Comment (koushikr) [reply .github/workflows/build-check.yml]: Done this with temurin, let me know if you want to do adopt. Also updated jacoco version to work with this to 0.8.8 (from 0.8.6)

Comment (koushikr) [reply .github/workflows/build-check.yml]: Done.

**Review by santanusinha — APPROVED** [review]

**Pull request merged**